### ISSUE TITLE
Mocha in browser not running tests

### ISSUE THREAD

**Issue by vladeck** [mochajs/mocha#3371]
I have tests written in **TypeScript** and transcompiled to single `.js` file using **Webpack**:

- Running `mocha tests.js` will work just fine.
- Running tests using **Karma** (`karma.conf.js` uses `mocha` framework, loads `tests.js`, browser is `PhantomJS`) will work just fine.

However, running tests by opening (in browser from local disk) provided `runner.html` will **NOT** work (I can see Mocha's HTML created under `<div id="mocha"></div>`, but no test results are shown):

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Tests</title>

    <link rel="stylesheet" href="../../../node_modules/mocha/mocha.css">
</head>
<body>

    <div id="mocha"></div>

    <script src="../../../node_modules/mocha/mocha.js"></script>
    <script src="../../../node_modules/chai/chai.js"></script>

    <script>
        mocha.setup("bdd")
    </script>

    <script src="./tests.js"></script>

    <script>
        mocha.run();
    </script>

</body>
</html>

```

![mocha_run_issue](https://user-images.githubusercontent.com/16443718/39577796-88e12bc2-4ee2-11e8-8d1a-42f4a839b0af.png)

Can anyone give me insight as to why?

**Comment by plroebuck** [comment]
No real experience with browser side of this project myself, but since you have the Inspector open...
* Anything shown in console tab?
* "mocha.js" actually loaded in resources tab?

Title of this says tests are not running, but DOM node "mocha-report" exists which suggests otherwise. Are your tests running (add `console.log()` to "tests.js" and maybe before/after `mocha.run()`)? Or is it that output is missing?

**Comment by plroebuck** [comment]
Where did it jump out of `mocha.run()` when you set your breakpoints?

**Comment by outsideris** [comment]
Could you test it with CDN like [here](https://github.com/outsideris/mocha/blob/e33fd51426960562d86e8215904b3a0f6fd0f106/docs/index.md#browser-specific-methods)?

**Comment by vladeck** [comment]
I'm going to close this as I have configured it to work flawlessly using Webpack + Karma. My guess (from limited exposure to this problem) is that the bundling of Mocha/Chai code using Webpack causes tests to not run - when bundled, there is no global `mocha` object (as I understand), thus including Mocha `.js` and calling `mocha.run` does nothing for the bundled Mocha code.

For anyone that wants to pursuit this further, this is what I did:

- built `.spec.ts` files into single `tests.js` (with no Mocha/Chai included code)
- bundled Mocha/Chai code into single `vendor.js` (My guess is that here lies the problem)

I can provide simple project if there is interest in tracking/understanding this issue.
